internal/abci/abci.go

```diff
@@ -672,6 +681,12 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 		a.halted.Store(true)
 	}
 
+	// Notify the migrator of the changeset
+	err = a.migrator.NotifyHeight(ctx, &blockCtx, a.db, writer)
```

Comment:
Pertaining to https://github.com/kwilteam/kwil-db/pull/887/files#r1702316468, here we've got this `io.ReadWriter` that's supposed to function as a pipe between the changeset collection from the pg logical replication stream and the storage routines.  However these two things two things cannot happen concurrently yet, with the current design.  That is, `NotifyHeight` does needs to be called after all the above is done, but before we call `Precommit` (and it starts writing to the `io.Writer`), another goroutine needs to be consuming the `io.Reader` and writing to disk.  i.e. the part that does `storeChangeset`